crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -1455,13 +1457,39 @@ impl<T: PoolTransaction> AllTransactions<T> {
     pub(crate) fn remove_transaction_by_hash(
         &mut self,
         tx_hash: &B256,
-    ) -> Option<(Arc<ValidPoolTransaction<T>>, SubPool)> {
+        update_descendant: bool,
+    ) -> Option<(Arc<ValidPoolTransaction<T>>, SubPool, Vec<PoolUpdate>)> {
         let tx = self.by_hash.remove(tx_hash)?;
         let internal = self.txs.remove(&tx.transaction_id)?;
         // decrement the counter for the sender.
         self.tx_decr(tx.sender_id());
         self.update_size_metrics();
-        Some((tx, internal.subpool))
+
+        if !update_descendant {
+            return Some((tx, internal.subpool, vec![]));
+        }
+
+        let mut updates = Vec::new();
+
+        for (id, tx) in self.descendant_txs_mut(tx.id()) {
+            let current_pool = tx.subpool;
+
+            tx.state.remove(TxState::NO_NONCE_GAPS);
```

Comment:
cool, this should actually do it, this is basically equivalent to reinserting them

Reply:
yes, similar to the operations in `insert_tx`